src/Fields/abstract_field.jl

```diff
@@ -79,9 +79,9 @@ interior(f::AbstractField) = f
 @propagate_inbounds ynode(i, j, k, ψ::AbstractField) = ynode(i, j, k, ψ.grid, instantiated_location(ψ)...)
 @propagate_inbounds znode(i, j, k, ψ::AbstractField) = znode(i, j, k, ψ.grid, instantiated_location(ψ)...)
 
-xnodes(ψ::AbstractField) = xnodes(ψ.grid, instantiated_location(ψ, 1))
-ynodes(ψ::AbstractField) = ynodes(ψ.grid, instantiated_location(ψ, 2))
-znodes(ψ::AbstractField) = znodes(ψ.grid, instantiated_location(ψ, 3))
+xnodes(ψ::AbstractField) = xnodes(ψ.grid, instantiated_location(ψ)[1])
+ynodes(ψ::AbstractField) = ynodes(ψ.grid, instantiated_location(ψ)[2])
+znodes(ψ::AbstractField) = znodes(ψ.grid, instantiated_location(ψ)[3])
```

Comment:
We need the three location versions here to be valid on all grids right?

Reply:
Even RectilinearGrid + ImmersedBoundaryGrid will need the three location versions to accomodate all possible immersed boundaries (even partial cells)

Reply:
Right. Good point!

Reply:
Also I think we want to propagate kwargs through this function so we can use `with_halo`